.github/workflows/buildtest.yml

```diff
@@ -33,10 +33,13 @@ jobs:
 
       - name: build start
         run: |
-          export ARCH=x86_64 CF="-Wsparse-error"
+          export ARCH=x86_64 CF="-Wsparse-error -Wsparse-all -Wno-bitwise-pointer -Wno-pointer-arith -Wno-typesign -Wnoshadow"
           bash kconfig/kconfig-sof-default.sh
           make modules_prepare
-          make -k M=sound/soc/sof C=2
+          make -k sound/soc/sof/ C=2
+          make -k sound/soc/intel/common/ C=2
+          make -k sound/soc/intel/boards/ C=2
+          make -k drivers/soundwire/ C=2
```

Comment:
If there are warnings in `sound/soc/sof/` then I'm 97% sure the others won't run. Is that what you want? If this is not what you want I see two possible solutions:

1. Separate github jobs. Pros: faster completion and smaller logs in each job. Cons: a lot of duplication in this `.yml` file (github does not support YAML anchors

2. `make -k sound/soc/sof/  sound/soc/intel/common ... C=2`

Reply:
this is what we do for other cases, I don't think it makes sense to continue compiling a different directory if the first one is already broken. The -k is already questionable, but I guess it's useful for people to correct all the errors in that directory.

Reply:
Sounds like a good trade-off, just wanted to make sure you were aware of this.